inflate.c

```diff
@@ -1050,6 +1050,18 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
                 if (INFLATE_NEED_CHECKSUM(strm) && state->wrap & 4) {
                     if (out) {
                         inf_chksum(strm, put - out, out);
+                    } else if ((!state->flags && strm->adler == ADLER32_INITIAL_VALUE)
+#ifdef GUNZIP
+                    || (state->flags && state->crc_fold.value == CRC32_INITIAL_VALUE)) {
```

Comment:
It would be great if there was a way we could clean up this if statement so it doesn't break like this.